classlib/src/main/java/org/teavm/classlib/java/lang/TInteger.java

```diff
@@ -45,7 +45,7 @@ public static String toString(int i, int radix) {
     }
 
     public static int hashCode(int value) {
-        return (value >>> 4) ^ (value << 28) ^ (value << 8) ^ (value >>> 24);
+        return value;
```

Comment:
What's the purpose of this change?

Reply:
Performance improvement.

Reply:
This *is* a performance improvement for running this particular function, but will it make object distribution in hashtable better? Do you have any CS papers with overview of quality of hashing functions? I'm no competent enough, but naively it seems that applying at least *some* transformation to value is better than not doing so.

Reply:
And also documentation states that hashcode should be equal to primitive int, not this strange permutation.

Reply:
Ah, I see. I wonder why Java guys added this requirement. In previous versions of Java particular implementation of Integer.hashCode was not fixed.